rules/legacy/src/Rector/Include_/AddTopIncludeRector.php

```diff
@@ -27,74 +24,34 @@
 final class AddTopIncludeRector extends AbstractFileSystemRector
 {
     /**
-     * @var string
+     * @var String_
      */
-    private $file = '';
+    private $autoloadFilePathString;
 
     /**
-     * @var ?Expr
-     */
-    private $fileExpression;
-
-    /**
-     * @var int
-     */
-    private $type = 0;
-
-    /**
-     * @var string[] to match
+     * @var string[]
      */
     private $patterns = [];
 
-    /**
-     * @var Parser
-     */
-    private $parser;
-
-    /**
-     * @var Standard
-     */
-    private $standard;
-
-    public function __construct(string $type = 'TYPE_INCLUDE', string $file = '"autoload.php"', array $match = [])
+    public function __construct(string $autoloadFilePath = '/autoload.php', array $match = [])
```

Comment:
The only relevant string is relative path to the file, `__DIR__` is added by default, so no need to add it in the string, parse it with parse, extract the expression and it again.

File string is enough here